runtime/include/map.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "lock.h"
+
+/* Simple K-V store based on The Practice of Programming by Kernighan and Pike */
+
+/* Bucket count is sized to be a prime that is approximately 20% larger than the desired capacity (6k keys) */
+#define MAP_BUCKET_COUNT 7907
+#define MAP_LOCK_STRIPES 17
```

Comment:
Great that you already instrumented the lock code (from before) to report total blocking time so that we can know if we have too much contention, and performance is paying. I think it is more than enough information to realize that there is a perf problem that requires debugging.

Nothing actionable: just a pat on the back of your past self.